docs/docs/dart_api/dataclass.md

```diff
@@ -1,23 +1,112 @@
-This works well for most cases: Drift knows  what columns your table has, and it can generate a simple class for all of that.
-In some cases, you might want to customize the generated classes though.
-For instance, you might want to add a mixin, let it extend another class or interface, or use other builders like
-`json_serializable` to customize how it gets serialized to json.
 
-As a solution, drift allows you to use your own classes as data classes for the database.
 
-## Using custom classes
+# Generated Dataclass
 
-To use a custom row class, simply annotate your table definition with `@UseRowClass`.
+Drift generates a dataclass for each table in your database. These dataclasses represent query results and come with built-in equality, hashing, and serialization support. They also include a `copyWith` method for easy modification.
+
+**Example:**
+
+For a `Users` table, Drift automatically generates a `User` dataclass. This dataclass is used for all read operations from the `Users` table, ensuring type-safe and structured data retrieval.
+
+{{ load_snippet('generated-dataclass','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+## Dataclass Name
+
+The dataclass name is derived from the table name.
+
+- If the name ends in `s`, the dataclass name will be the name with `s` removed.
+    - Example: `Users` -> `User`
+- Otherwise, the dataclass name will be the name with `Data` appended.
+    - Example: `UserInfo` -> `UserInfoData`
+
+
+To use a custom name use the `@DataClassName` annotation.
+
+**Example:**
+
+{{ load_snippet('data-class-name','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+## Json serialization
+
+### Key names
 
+When serializing to json, the generated dataclass will use the column name in `snake_case` for the json keys.
+
+**Example:**
+
+{{ load_snippet('default-json-keys','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+```json
+{
+  "id": 1,
+  "title": "Todo 1",
+  "created_at": "2024-02-29T12:00:00Z"
+}
+```
+
+### Custom json keys
+
+To use a custom name for JSON serialization, use the `@JsonKey` annotation.
+
+**Example:**
+
+{{ load_snippet('custom-json-keys','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+```json
+{
+  "id": 1,
+  "title": "Todo 1",
+  "created": "2024-02-29T12:00:00Z"
+}
+```
+
+If you prefer to use the actual column name in SQL as the JSON key, set `use_sql_column_name_as_json_key` to `true` in the `build.yaml` file.
+
+```yaml title="build.yaml"
+targets:
+  $default:
+    builders:
+      drift_dev:
+        options:
+          use_sql_column_name_as_json_key : true
+```
+For more details on customizing column names in SQL, refer to the [column name](tables.md#column-names) documentation.
+
+## Companions
+
+In addition to the generated dataclass representing a complete row, Drift also generates a companion object for each table, which represents a partial row and can be used to update existing rows.
```

Comment:
I think this is a pretty short explanation for an important subject. I think we should take something [like this](https://github.com/simolus3/drift/blob/6d31b37ae52ffa23f21e3e075b25ae7210790dc9/docs/docs/dart_api/writes.md?plain=1#L42-L56) as a better reference on how to introduce companions.

We could also mention that companions are not as relevant with the manager API as their structure is reflected by the callbacks for creating and updating rows. (similar to the annotation below)